2015/proposals/li-yanghao.md

```diff
@@ -0,0 +1,83 @@
+# Add taxonomic name resolution to the EcoData Retriever to facilitate data-intensive approaches to ecology
+
+## Abstract
+
+The EcoData Retriever is a Python based tool for automatically downloading, cleaning up, restructuring ecological data files into standard formats, and then storing them in your choice of database management systems. It does the hard work of data munging so that scientists can focus on doing science.
+
+This project focuses on the problem that the names of species in ecological data are constantly being redefined. By automating reconciliation of different species names as part of the process of accessing the data in the first place, it will become much easier for users to combine diverse datasets for specific scientific research.
+
+## Technical Details
+
+This project would build a system for sending a species name to one or more of the taxonomic name resolution services and determine the best name to appear in the generated databases. So the project mainly contains three components: accessing the resolution services, determining the election strategy and updating the user interfaces.
+
+1. **Accessing the resolution services.** This could be built by using [pytaxize](https://github.com/sckott/pytaxize) to resolve species names with specific parameters. Current `pytaxize` supports resolving names by some services, such as [Global Names Resolver](http://resolver.globalnames.org/). We could also build out `pytaxize` to support more resolution services and have the Retriever use it.
+
+2. **Determining the election strategy.** Since each resolution service may return many candidate names, we need to choose a best name according to the returned scores of services and some qualities of the candidate names. We also need to determine a few default standard data sources since it would be much faster than searching all data sources. For example, there are [181 sources](http://resolver.globalnames.org/data_sources) in `GNR`. My current idea is simple, which is to choose the name with highest average score from all data sources and the score also must be higher than a threshold. I will continue to talk with mentors to determine the final election strategy.
```

Comment:
1. Names are not always spelled correctly. I think the return results of resolution services could correct these at most cases. If not, maybe we could use some other information to describe the quality of the name, such as if dataset type matches species type. 
2. The same name for a plant and an animal. I'm afraid I don't get the point of this problem. When this name is spelled incorrectly, I think the services would return the correct name no matter they regard it as a plant or an animal.